hw2/03-count.js

```diff
@@ -1,3 +1,24 @@
-// Add your code here
+//Return the source word, with html to highlight it if it matches the target word
+const replaceWithHighlightHTML = (source, target) => {
+  return source === target
+    ? `<mark class='bg-warning'>${target}</mark>`
+    : source;
+};
 
-input.addEventListener('keydown', handleKeyDown);
+//Highlight words in the paragraph based on user input
+const handleKeyUp = () => {
+  const input_value = document
+    .getElementById("userInput")
+    .value.trim()
+    .toLowerCase();
+  const search_element = document.getElementById("toSearch");
+  const search_arr = search_element.textContent
+    .toLowerCase()
+    .split(RegExp("\\b"));
+
+  search_element.innerHTML = search_arr
+    .map((word) => replaceWithHighlightHTML(word, input_value))
+    .join("");
+};
+
+document.getElementById("userInput").addEventListener("keyup", handleKeyUp);
```

Comment:
Based on the resources provided it mentions **keydown** event not keyup event.

Reply:
Getting low on time to finish this so I had to drop this fix (since I'm still unsure how to make it not lag one keystroke behind on keydown). From a user perspective the difference is unnoticeable so I made this a low priority.

Reply:
"since I'm still unsure how to make it not lag one keystroke behind on keydown", From what I understood from that statement is that keydown event is pressed when the key is held down, you are experiencing the lag because the key is not completely pressed and released. For that you can attach a "Enter key" and then read the value from the input field.

I only mentioned this, as keydown was in resource, and I understand the low on time you have to finish this.